### PR TITLE
Replace validation error messages with official copy

### LINKED ISSUE #634: `hasValue` in the `ApplicantQuestion` subclasses don't seem to WAI
We decided on a convention of storing `null` for "never answered" pieces of data, and `""` (the empty string) for "answered with a blank value".

However, it seems like the underlying spring form data binder binds empty inputs as `null` (at least that's what I guess is happening).

I ran into this issue while implementing min and max length validations for text questions. When there is no value entered in the applicant form, our code evaluates `hasValue()` to `false`, so we end up not running our `getQuestionErrors` validations.

Needs more investigation to conclude the root cause and obviously to fix the issue as well.

### PR DESCRIPTION
**Pull request by natsid**
### Description
* Replace the placeholder validation error messages with the official copy from [this doc](https://docs.google.com/document/d/1chNdJViEqadU8pBMWXAXc4dOmsIrplDC6vva0ScoSXM/edit?usp=sharing).
* Move the address question validation error messages to the `ValidationErrorMessage` class.
* Replace `hasValue` with `isAnswered` in `TextQuestion` (bug fix; will do the same fix for `NumberQuestion` and any other types that need it in a follow-up PR to fully fix #634).

### Checklist
- [x] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### Issue(s)
Polish work toward #377 
Works toward #634

### CODE REVIEW

File: universal-application-tool-0.0.1/test/services/applicant/question/AddressQuestionTest.java [hunk 1]
```diff
@@ -95,21 +95,22 @@ public void withInvalidApplicantData_missingRequiredFields() {
 
     assertThat(addressQuestion.hasTypeSpecificErrors()).isTrue();
     assertThat(addressQuestion.getStreetErrors())
-        .contains(ValidationErrorMessage.create("Street is required."));
+        .contains(ValidationErrorMessage.create("Please enter valid street name and number."));
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/test/services/applicant/question/AddressQuestionTest.java:98]: question: Should we use the message constants here too?

File: universal-application-tool-0.0.1/test/services/applicant/question/AddressQuestionTest.java [hunk 2]
```diff
@@ -95,21 +95,22 @@ public void withInvalidApplicantData_missingRequiredFields() {
...
 
   @Test
-  public void withInvalidApplicantData_invalidZipCode() {
+  @Parameters({"not a zip code", "123456789", "123ab"})
+  public void withInvalidApplicantData_invalidZipCode(String zipValue) {
     applicantData.putString(addressQuestionDefinition.getStreetPath(), "123 A St");
     applicantData.putString(addressQuestionDefinition.getCityPath(), "Seattle");
     applicantData.putString(addressQuestionDefinition.getStatePath(), "WA");
-    applicantData.putString(addressQuestionDefinition.getZipPath(), "not a zip code");
+    applicantData.putString(addressQuestionDefinition.getZipPath(), zipValue);
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/test/services/applicant/question/AddressQuestionTest.java:113]: praise: good catch!

File: universal-application-tool-0.0.1/test/services/applicant/question/TextQuestionTest.java [hunk 3]
```diff
@@ -93,9 +93,9 @@ public void withMinAndMaxLength_withValidApplicantData_passesValidation(String v
   @Parameters({
     // TODO(https://github.com/seattle-uat/civiform/issues/634): Uncomment the following line once
     //  this bug is fixed.
```
Comment (carolinedanzi) [inline universal-application-tool-0.0.1/test/services/applicant/question/TextQuestionTest.java:95]: suggestion (blocking): please remove comment if no longer applicable

**Review by carolinedanzi — APPROVED** [review]
This looks nice!

File: universal-application-tool-0.0.1/test/services/applicant/question/AddressQuestionTest.java (hunk 1 above)
Comment (natsid) [reply universal-application-tool-0.0.1/test/services/applicant/question/AddressQuestionTest.java:98]: Good call!

File: universal-application-tool-0.0.1/test/services/applicant/question/TextQuestionTest.java (hunk 3 above)
Comment (natsid) [reply universal-application-tool-0.0.1/test/services/applicant/question/TextQuestionTest.java:95]: yay - thanks for catching this!

**Pull request merged**